test/splitter.js

```diff
@@ -0,0 +1,206 @@
+            "The first recipient is the same as the second recipient"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("Transaction reverts if the sender's address is also a recipient", async () => {
+        await truffleAssert.reverts(
+            splitter.splitDeposit(sender, recipientTwo, {from: sender, value: 4}),
+            "The sender cannot also be a recipient"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("SplitDeposit is pausable", async () => {
+
+        await splitter.pause({from: contractOwnerAddress});
+
+        await truffleAssert.reverts(
+            splitter.splitDeposit(recipientTwo, recipientOne,{from: sender, value: 4}),
+            "Pausable: paused"
+        );
+        checkEventNotEmitted();
+    });
+
+    it("SplitDeposit is unpausable", async () => {
+
+        await splitter.pause({from: contractOwnerAddress});
+        await splitter.unpause({from: contractOwnerAddress});
+
+        const txObj = await splitter.splitDeposit(recipientTwo, recipientOne,{from: sender, value: 5});
+
+        truffleAssert.eventEmitted(txObj, 'Deposit');
+    })
+
+
+    it('Second recipient can successfully withdraw 2 wei', async () => {
+
+        const depositAmount = toBN(5);
+        const withDrawAmount = toBN(2);
+        
+        // Deposit into the contract
+        await splitter.splitDeposit(recipientOne, recipientTwo, {from: sender, value: depositAmount});
+
+        // Take a snapshot of the second recipient's ETH balance
+        const initRecipientTwoEthBalance = toBN(await web3.eth.getBalance(recipientTwo));
+
+        // Recipient withdraws
+        const txObj = await splitter.withdraw(withDrawAmount, {from: recipientTwo});
+        const cost = await getGasCost(txObj);
+
+        // Get the recipient's new ETH and contract balances
+        const recipientTwoEthBalance = toBN(await web3.eth.getBalance(recipientTwo));
+        const recipientTwoOwed = toBN(await splitter.balances(recipientTwo));
+
+        // Calculate the expected new ETH and contract balances
+        const expectedRecipientTwoEthBalance = initRecipientTwoEthBalance.sub(cost).add(withDrawAmount).toString(10);
+        const expectedRecipientTwoOwed = "0";
+
+        assert.strictEqual(recipientTwoEthBalance.toString(10), expectedRecipientTwoEthBalance);
+        assert.strictEqual(recipientTwoOwed.toString(10), expectedRecipientTwoOwed);
+
+        truffleAssert.eventEmitted(txObj, "WithDraw", (ev) => {
+            return  ev.withdrawer === recipientTwo &&
+                    ev.amount.toString(10) === "2";
+        }, 'TestEvent should be emitted with correct parameters');
+
+    });
+
+    it("Second recipient attempts to withdraw more than is available in the balance", async () => {
+
+        withDrawAmount = toWei(toBN(5), "ether");
+
+        await truffleAssert.reverts(
+            splitter.withdraw(withDrawAmount, {from: recipientTwo}),
```

Comment:
As it happens, at this stage, the recipient has 0 owed. The contract also has 0. 
A more meaningful test would see the contract balance of, say 5, owed 2 to the recipient, and trying to withdraw 3, which would mean potentially stealing from others.